types/node-zendesk/index.d.ts

```diff
@@ -0,0 +1,659 @@
+     */
+    interface UpdateModel {
+        comment?: Comments.CreateModel;
+        solved?: boolean;
+        additional_collaborators?: ZendeskID[] | string[] | Collaborator[];
+    }
+
+    /**
+     * @see {@link https://developer.zendesk.com/rest_api/docs/support/requests#json-format|Zendesk Requests JSON Format}
+     */
+    interface ResponseModel extends AuditableModel {
+        readonly url: string;
+        readonly subject: string;
+        readonly description: string;
+        readonly status: Tickets.Status;
+        readonly priority: Tickets.Priority | null;
+        readonly type: Tickets.Type | null;
+        readonly custom_fields: Tickets.Field[] | null;
+        readonly fields: Tickets.Field[] | null;
+        readonly organization_id: ZendeskID | null;
+        readonly requester_id: ZendeskID;
+        readonly assignee_id: ZendeskID | null;
+        readonly group_id?: ZendeskID | null;
+        readonly collaborator_ids: ZendeskID[];
+        readonly email_cc_ids: ZendeskID[];
+        readonly via: Tickets.Via;
+        readonly is_public: boolean;
+        readonly due_at: string | null;
+        readonly can_be_solved_by_me?: boolean;
+        readonly solved?: boolean;
+        readonly ticket_form_id?: number | null;
+        readonly recipient: string | null;
+        readonly followup_source_id: string | null;
+    }
+
+    interface RequesterAnonymous {
+        name: string;
+        email?: string;
+        locale_id?: ZendeskID;
+    }
+
+    interface Collaborator {
+        name?: string;
+        email: string;
+    }
+
+    interface CreatePayload {
+        readonly request: CreateModel;
+    }
+
+    interface UpdatePayload {
+        readonly request: UpdateModel;
+    }
+
+    interface ResponsePayload {
+        readonly request: ResponseModel;
+    }
+
+    interface ListPayload extends PaginablePayload {
+        readonly requests: ReadonlyArray<ResponseModel>;
+    }
+
+    namespace Comments {
+        interface CreateModel {
+            url?: string;
+            request_id?: number;
+            body?: string;
+            html_body?: string;
+            public?: boolean;
+            author_id?: ZendeskID;
+            uploads?: ReadonlyArray<string>;
+        }
+
+        // TODO: support VoiceComment model @jgeth 2019-01-22
+        interface ResponseModel extends TemporalModel {
+            readonly url: string;
+            readonly type: Type;
+            readonly request_id: number;
+            readonly body: string;
+            readonly html_body: string;
+            readonly plain_body: string;
+            readonly public: boolean;
+            readonly author_id: ZendeskID;
+            readonly attachments: ReadonlyArray<Attachments.Model>;
+            readonly via?: Tickets.Via;
+            readonly metadata?: Tickets.Comments.Metadata;
+        }
+
+        enum Type {
```

Comment:
Is this an enum that exists at runtime? I can't find it within https://github.com/blakmatrix/node-zendesk

If it isn't, it needs to either be a `const enum` or a type alias.

Reply:
This comment probably applies to all uses of `enum`

Reply:
enums have been replaced with type aliases consisting of unions of literals.